pkg/cmd/util/route.go

```diff
@@ -54,6 +54,10 @@ func UnsecuredRoute(kc kclientset.Interface, namespace, routeName, serviceName,
 		},
 	}
 
+	if useTargetPort {
+		route.Spec.Port = getTargetPort(svc)
+	}
+
 	// If the service has multiple ports and the user didn't specify --port,
 	// then default the route port to a service port name.
 	if len(port.Name) > 0 && len(portString) == 0 {
```

Comment:
You should do the check in this condition:

if ( len(port.Name) > 0 || useTargetPort ) && len(portString) == 0 {